subgraphs/rocket-pool/src/checkpoints/stakerUtilities.ts

```diff
@@ -0,0 +1,357 @@
+import { Address, ethereum } from "@graphprotocol/graph-ts";
+import {
+  NetworkStakerBalanceCheckpoint,
+  Staker,
+  StakerBalanceCheckpoint,
+  RocketETHTransaction,
+  RocketPoolProtocol,
+} from "../../generated/schema";
+import { TransactionStakers } from "../models/transactionStakers";
+import { StakerBalance } from "../models/stakerBalance";
+import { BigInt } from "@graphprotocol/graph-ts";
+import { rocketPoolEntityFactory } from "../entityFactory";
+import { ONE_ETHER_IN_WEI } from "../constants/generalConstants";
+import { ZERO_ADDRESS_STRING } from "../constants/contractConstants";
+import { generalUtilities } from "./generalUtilities";
+
+class StakerUtilities {
+  /**
+   * Returns the hexadecimal address representation.
+   */
+  public extractStakerId(address: Address): string {
+    return address.toHexString();
+  }
+
+  /**
+   * Checks if there is already an indexed network staker balance checkpoint for the given event.
+   */
+  public hasNetworkStakerBalanceCheckpointHasBeenIndexed(
+    protocol: RocketPoolProtocol,
+    event: ethereum.Event
+  ): boolean {
+    // If this specific event has been handled, then return true.
+    if (
+      NetworkStakerBalanceCheckpoint.load(
+        generalUtilities.extractIdForEntity(event)
+      ) !== null
+    )
+      return true;
+
+    // No indexed protocol means there is no latest network staker balance checkpoint.
+    if (protocol === null) return false;
+
+    /*
+      Retrieve the latest network balance checkpoint.
+      If there is none at the moment, return false because this hasnt been handled yet.
+    */
+    const latestNetworkStakerBalanceCheckpointId =
+      protocol.lastNetworkStakerBalanceCheckPoint;
+    if (latestNetworkStakerBalanceCheckpointId === null) return false;
+    const latestNetworkStakerBalanceCheckpoint =
+      NetworkStakerBalanceCheckpoint.load(
+        latestNetworkStakerBalanceCheckpointId
+      );
+    if (
+      latestNetworkStakerBalanceCheckpoint === null ||
+      latestNetworkStakerBalanceCheckpoint.blockTime == BigInt.fromI32(0)
+    )
+      return false;
+
+    // Get the date of the network staker balance event candidate and the latest network staker balance checkpoint.
+    const dateOfNewNetworkStakerBalanceCheckpoint = new Date(
+      event.block.timestamp.toI32() * 1000
+    );
+    const dateOfLatestNetworkStakerBalanceCheckpoint = new Date(
+      latestNetworkStakerBalanceCheckpoint.blockTime.toI32() * 1000
+    );
+
+    // If the latest network staker balance checkpoint and the candidate match in terms of day/month/year, then return false.
+    return (
```

Comment:
same ting

Reply:
See above comment. not sure what todo about this but my instinct is don't change it if it currently works?